backend/routers/georeferencing.py

```diff
@@ -1,32 +1,190 @@
-from modules.georeferenser import add_gcp
-from modules.models import Point
-#Router handles all requests to the georeferencing API
-router = APIRouter()
+from fastapi import APIRouter, File, UploadFile, HTTPException
+from fastapi.responses import FileResponse
+from modules.models import *
+from typing import List
+from localrepository import Projects
 
+#Router handles all requests to the georeferencing API/ depending on the singelton repository of app in main.py
+#The router is included in the main.py and the routes are added to the router
+router = APIRouter(
+    prefix="/project",
+    tags=["georeferencing"],
+    responses={404: {"description": "Not found"},
+                415: {"description": "Unsupported media type"},
+                400: {"description": "Bad request"}
+               },
+)
 
-#route for georeferncing the gtiff uses the add_gcp function from georeferencer.py
-#function parameters: image: bytes, points: list[Point]
+#adding the singleton repository to the router to be able to access the projects
+router.projects = Projects()
+Repo = router.projects
+
+#GeorefProject path: /georef/project
+#route to create and or get georef a project id
 @router.post("/")
-async def georeference(file: UploadFile = File(...), points: list[Point] = Form(...)):
-    #check if file is a geotiff
-    if file.content_type != "image/tiff":
-        raise HTTPException(status_code=400, detail="File is not a geotiff")
-    #check if points is a list
-    if not isinstance(points, list):
-        raise HTTPException(status_code=400, detail="Points is not a list")
-    #check if points is a list of Point models
-    if not all(isinstance(point, Point) for point in points):
-        raise HTTPException(status_code=400, detail="Points is not a list of Point models")
-    #add gcp to image
-    gdf = add_gcp(file.file, points)
-    #check if gdf is None
-    if gdf is None:
-        raise HTTPException(status_code=400, detail="Not enough points")
-    #return as gtiff file
-    return gdf.to_file("georeferenced.tif", driver="GTiff")
-
-#path: eks http://localhost:8000/georef/1
-@router.get("/{id}", tags=["georeferencing"])
-async def get_point(id: int):
-    return {"message": "Hello World"}
+async def georefProject(project: Project):
+    try:
+        Repo.addProject(project)
+        #get the created project and return it's id
+        return {"id": project.id}
+    except:
+        raise HTTPException(status_code=400, detail="Project could not be created")
+
+#route update project details
+@router.put("/{projectId}")
+async def updateProject(projectId: int):
+    #try to find the project by id and update it
+    try:
+        project = Repo.getProject(projectId)
+        Repo.updateProject(projectId, project)
+        return {"id": projectId}
+    except Exception as e:
+        raise HTTPException(status_code=404, detail=str(e))
+
+
+#route to delete a project
+@router.delete("/{projectId}")
+async def deleteProject(projectId: int):
```

Comment:
Error, not found: "detail": "[WinError 2] Systemet finner ikke angitt fil:

Reply:
Fixed with [3ef32c5](https://github.com/TheAtlasRepository/ImgPDF2Map/pull/29/commits/3ef32c588d5a430c2f0006d2ae80972730916865)